File: ios/Classes/RefinerClass.swift
import Flutter
import RefinerSdk

public class RefinerClass {
    static func initRefiner(call: FlutterMethodCall){
        let projectId = call.arguments as! String;
        do {
            try Refiner.instance.initialize(projectId: projectId, enableDebugMode: true)
        } catch {
            print(error.localizedDescription)
        }
    }

    static func identifyUser(call: FlutterMethodCall){
        let dict = call.arguments as! Dictionary<String, Any>;
        let userId = dict["userId"]! as! String;
        let userTraits = dict["userTraits"] as! Dictionary<String, String>;
        do { try
            Refiner.instance.identifyUser(
            userId: userId,
            userTraits: userTraits); 
        } catch {
            print(error.localizedDescription);
        }
    }


    static func trackEvent(call: FlutterMethodCall){
        let eventName = call.arguments as! String;
        do {
            try Refiner.instance.trackEvent(name: eventName)
        } catch {
            print(error.localizedDescription)
        }
    }

    static func trackScreen(call: FlutterMethodCall){
        let screenName = call.arguments as! String;
        do {
            try Refiner.instance.trackScreen(name: screenName)
        } catch {
            print(error.localizedDescription)
        }
    }

    static func resetUser() {
        do {
            try Refiner.instance.resetUser();
        } catch {
            print(error.localizedDescription)
        }
    }
}

File: android/src/main/kotlin/br/com/intelipost/refiner_class/RefinerClass.kt
package br.com.intelipost.refiner_class

import androidx.annotation.NonNull

import android.content.Context
import io.flutter.plugin.common.MethodCall
import io.refiner.Refiner
import io.refiner.RefinerConfigs
import android.util.Log

class RefinerClass {
    companion object {
        @JvmStatic
        fun initRefiner(@NonNull call: MethodCall, @NonNull context: Context) {
            var projectId = call.arguments() as String?;
            try {
                Refiner.initialize(
                    context = context,
                    refinerConfigs = RefinerConfigs(
                        projectId = projectId!!,
                        enableDebugMode = true
                    )
                );
            } catch (e: Exception) {
                Log.e("REFINER", e.printStackTrace().toString());
            }
        }

        @JvmStatic
        fun identifyUser(@NonNull call: MethodCall) {
            var arguments = call.arguments() as HashMap<String, String>?;
            var userTraits = arguments!!.get("userTraits") as HashMap<String, String>?;
            var newMap = LinkedHashMap<String, Any>(userTraits);
            try {
                Refiner.identifyUser(
                    userId = arguments!!.get("userId") as String,
                    userTraits = newMap
                );
            } catch (e: Exception) {
                Log.e("REFINER", e.printStackTrace().toString());
            }
        }


        @JvmStatic
        fun trackEvent(@NonNull call: MethodCall) {
            var eventName = call.arguments() as String?;
            try {
                Refiner.trackEvent(eventName = eventName!!);
            } catch (e: Exception) {
                Log.e("REFINER", e.printStackTrace().toString());
            }
        }

        @JvmStatic
        fun trackScreen(@NonNull call: MethodCall) {
            var screenName = call.arguments() as String?;
            try {
                Refiner.trackScreen(screenName = screenName!!);
            } catch (e: Exception) {
                Log.e("REFINER", e.printStackTrace().toString());
            }
        }

        @JvmStatic
        fun resetUser() {
            try {
                Refiner.resetUser();
            } catch (e: Exception) {
                Log.e("REFINER", e.printStackTrace().toString());
            }
        }
    }
}